TradeItIosEmsApi/UserAgent.m

```diff
@@ -0,0 +1,43 @@
+#import "UserAgent.h"
+#include <UIKit/UIKit.h>
+#include <sys/sysctl.h>
+
+@implementation UserAgent
+
+
++ (NSString *)getUserAgent {
+    NSDictionary<NSString *, id> *bundleDict = [[NSBundle mainBundle] infoDictionary];
+    NSString *appName = [bundleDict valueForKey:@"CFBundleName"];
+    NSString *appVersion = [bundleDict valueForKey:@"CFBundleShortVersionString"];
+    
+    NSString * appDescriptor = [NSString stringWithFormat:@"%@/%@", appName, appVersion];
+    
+    UIDevice *device = [UIDevice currentDevice];
+    NSString* systemVersion = [device systemVersion];
+    
+    NSString * osDescriptor = [NSString stringWithFormat:@"%@ %@", @"iOS", systemVersion];
+    
+    NSString * hardwareString = [self getSysInfoByName:"hw.model"];
+    
+    NSDictionary<NSString *, id> *bundleSDK2 = [[NSBundle bundleWithIdentifier:@"TradeIt.TradeItIosTicketSDK2"] infoDictionary];
+    NSString *sdkName = [bundleSDK2 valueForKey:@"CFBundleName"];
```

Comment:
You can see how I'm doing that here: https://github.com/tradingticket/TradeItIosTicketSDK2/commit/0d6ce0af19801cbcdbfbb4f37cd098bea877e669#diff-3f0b22de9c3bda5ddd65506c5f844b38 Look for the references to bundle and the import statements at the top.